### PR TITLE
[Doc] [BugFix] Add more docs about sync/async materialized views

### PR DESCRIPTION
**Pull request by LiShuMing**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/reference/System_variable.md [hunk 1]
```diff
@@ -174,6 +174,72 @@ SELECT /*+ SET_VAR
...
+
+### materialized_view_subuqery_text_match_max_count
+
+指定检查一个查询中，递归匹配子查询与定义的物化视图匹配的最大次数。
+
+默认值： 4
+
+### enable_force_rule_based_mv_rewrite(3.3 and later)
+
+* Description: Whether to rewrite input query in optimizer's CRO phase even for queries with multi tables.
```
Comment (ABingHuang) [inline docs/zh/reference/System_variable.md:216]: use chinese?

File: docs/zh/reference/System_variable.md [hunk 2]
```diff
@@ -174,6 +174,72 @@ SELECT /*+ SET_VAR
...
+默认值： true
+
+### enable_view_based_mv_rewrite(3.2 and later)
+
+* Description: Whether to enable view based rewrite or not. If true, treat logical view as unified node to rewrite rather than inlining it to be better for rewrite.
```
Comment (ABingHuang) [inline docs/zh/reference/System_variable.md:221]: use chinese

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 3]
```diff
@@ -1185,6 +1185,10 @@ public class Config extends ConfigBase {
     @ConfField(mutable = true, comment = "whether to collect metrics for materialized view by default")
     public static boolean enable_materialized_view_metrics_collect = true;
 
+    @ConfField(mutable = true, comment = "whether to enable text based rewrite by default, if true it will " +
+               "build ast tree in materialized view initialization")
+    public static boolean enable_materialized_view_text_based_rewrite = true;
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1190]: why use a config not a SessionVariable?

File: docs/en/administration/management/FE_configuration.md [hunk 4]
```diff
@@ -1210,6 +1210,24 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+##### enable_materialized_view_metrics_collect
+
+- Default: true
+- Type: Boolean
+- Unit: -
+- Is mutable: Yes
+- Description: whether to collect metrics for materialized view by default.
+- Introduced in: v3.1.11
+
+##### enable_materialized_view_text_based_rewrite
```
Comment (kangkaisen) [inline docs/en/administration/management/FE_configuration.md:1222]: why this is a config ,not session variable?

**Comment by LiShuMing** [comment]
@mergify rebase

**Review by ABingHuang — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java (hunk 3 above)
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1193]: then session variable is enough? In which case we should use config other than session variable of enable_materialized_view_text_match_rewrite?

**Review by ABingHuang — CHANGES_REQUESTED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]

**Pull request merged**